game/unicode.hh

```diff
@@ -2,30 +2,57 @@
 
 #include <iostream>
 #include <map>
+#include <memory>
+#include <mutex>
 #include <string>
 #include <unicode/errorcode.h>
 #include <unicode/tblcoll.h>
 #include <unicode/uclean.h>
+#include <unicode/ucnv.h>
 #include <unicode/ucsdet.h>
 
 typedef std::map<std::string, std::string> songMetadata;
 
+class UnicodeString;
+
+struct Converter {
+	/// Constructor/Destructor
+	Converter(std::string const& codepage);
+	Converter(Converter&& c) noexcept;
+	Converter(Converter& c) = delete;
+	Converter(Converter const& c) = delete;	
+	~Converter() {};
+
+	icu::UnicodeString convertToUTF8(std::string_view sv); ///< Do the actual conversion.
+
+	private:
+	std::string m_codepage;
+	std::unique_ptr<UConverter, decltype(&ucnv_close)> m_converter;
+	void reset(); ///< UConverters are state machines and need to be reset after each use.
```

Comment:
Put it in a new private section or move it up. I think members and functions shouldn't be mixed.

Reply:
Actually I forgot I can probably delete this function because the converter is automatically reset when used.

Reply:
OK